src/main/scala/analysis/data_structure_analysis/LocalPhase.scala

```diff
@@ -220,6 +180,27 @@ class LocalPhase(proc: Procedure,
     node.cells(0)
   }
 
+
+
+  def multiAccess(lhsOrValue: Cell, pointer: Cell, size: Int): Unit = {
+    // TODO there should be another check here to see we cover the bytesize
+    // otherwise can fall back on expanding
+
+    val diff = pointer.offset - lhsOrValue.offset
+    val startPointerOffset = pointer.offset
+    val lhsNode = lhsOrValue.node.get
+
+
+    val pointers = pointer.node.get.cells.filter((offset, _) => offset >= startPointerOffset && offset < startPointerOffset + size).toSeq.sortBy((offset, cell) => offset)
+    for (((offset, cell), i) <- pointers.zipWithIndex) {
+      val lhs = lhsNode.addCell(offset - diff, 0) // todo check if 0 is the right size
+      graph.find(cell).growSize(if i < pointers.size - 1 then (pointers(i + 1)._1 - offset - diff).toInt else (size - (offset - diff)).toInt)
+      val res = graph.mergeCells(lhs, graph.adjust(graph.find(cell).getPointee))
+       graph.handleOverlapping(res)
+    }
+  }
```

Comment:
This method could really use a comment explaining what it is doing, as it is very convoluted as-is